packages/matchbox/src/components/Select/Select.js

```diff
@@ -1,120 +1,156 @@
-     */
-    options: PropTypes.arrayOf(
-      PropTypes.oneOfType([
-        PropTypes.shape({
-          value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-          label: PropTypes.string.isRequired
-        }),
-        PropTypes.string.isRequired,
-        PropTypes.number.isRequired
-      ])
-    ).isRequired,
-    disabled: PropTypes.bool,
-    required: PropTypes.bool,
-    label: PropTypes.string,
-    helpText: PropTypes.node,
-    error: PropTypes.string,
-    errorInLabel: PropTypes.bool,
-    onChange: PropTypes.func,
-    onFocus: PropTypes.func,
-    onBlur: PropTypes.func
-  };
-
-  render() {
-    const {
-      id,
-      options,
-      label,
-      helpText,
-      placeholder,
-      placeholderValue,
-      disabled,
-      required,
-      error,
-      errorInLabel,
-      ...rest
-    } = this.props;
-
-    const setClasses = classnames(
-      styles.Select,
-      error && styles.error
-    );
+function Options({ options, placeholder, placeholderValue }) {
+  let combined = options;
 
-    const inputClasses = classnames(
-      styles.Input,
-      disabled && styles.disabled
-    );
+  if (placeholder) {
+    combined = [{ label: placeholder, value: placeholderValue, disabled: true }, ...combined];
+  }
 
-    const dropdownClasses = classnames(
-      styles.Dropdown,
-      !label && styles.labelHidden
-    );
+  if (!combined && !combined.length) {
+    return null;
+  }
+
+  return (
+    <>
+      {combined.map((option, key) => (
+        <Option option={option} key={key} />
+      ))}
+    </>
+  );
+}
 
-    let combined = options;
+const SelectBox = props => {
+  return (
+    <Box
+      as="select"
+      disabled={props.disabled}
+      width="100%"
+      border={props.hasError ? `1px solid ${tokens.color_red_700}` : '400'}
+      borderRadius="100"
+      bg={props.disabled ? 'gray.200' : 'white'}
+      pl="400"
+      pr="600"
+      lineHeight="2.5rem"
+      height="2.5rem"
+      color="gray.900"
+      {...props}
+    />
+  );
+};
 
-    if (placeholder) {
-      combined = [ { label: placeholder, value: placeholderValue, disabled: true }, ...combined ];
-    }
+const StyledSelect = styled(SelectBox)`
+  ${select}
+`;
 
-    const optionMarkup = combined && combined.length
-      ? combined.map((option, key) => <Option option={option} key={key} />)
-      : null;
+const StyledChevron = styled(KeyboardArrowDown)`
+  ${chevron}
+`;
 
-    const requiredIndicator = required
-      ? ' *'
-      : '';
+const StyledWrapper = styled(Box)`
+  {margin}
+`;
 
-    const labelMarkup = (
-      <Label id={id} label={`${label}${requiredIndicator}`}>
-        {error && errorInLabel && <Error className={styles.InlineError} wrapper='span' error={error} />}
-      </Label>
-    );
+function Select(props) {
+  const {
+    id,
+    options,
+    label,
+    helpText,
+    placeholder,
+    placeholderValue,
+    disabled,
+    required,
```

Comment:
This prop could also render the `required` attribute on the `<select/>` - an easy a11y win. Easy to add to any other `<input/>` element as well.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required